src/curve/stable/StableMintBurn.sol

```diff
@@ -140,8 +142,10 @@ contract StableMintBurn is ReservoirPair {
 
         uint256 lBalance0 = _totalToken0();
         uint256 lBalance1 = _totalToken1();
-        lastInvariant = uint192(_computeLiquidity(lBalance0, lBalance1));
-        lastInvariantAmp = _getCurrentAPrecise();
+        if (feeOn) {
+            lastInvariant = uint192(_computeLiquidity(lBalance0, lBalance1));
+            lastInvariantAmp = _getCurrentAPrecise();
+        }
```

Comment:
We write lastInvariant here as well as in `_mintFee` which seems like pretty confusing/ugly code organization. On a slightly side note, what is the reason we can't just always write the last invariant?

Reply:
This approach is following what we do for `ConstantProductPair`. Uni v2 has this check before writing `kLast`. I posit that they do this to save 1 `SLOAD` if it ain't necessary.

Alright so, after thinking about this for quite a bit, it seems to me that always writing `lastInvariant` at the end of `mint`/`burn` is not a problem. But the `if (_dLast != 0)` check and `else if (lastInvariant != 0) lastInvariant = 0;` check is essential to cater for the case of turning the platform fee on and off.  

Given that we expect all our pairs to have some form of platform fee from launch, I propose we get rid of this `feeOn` check for both curves.

Reply:
Sounds good, i like the simplicity

Reply:
fixed